### PR TITLE
Fixes the Unicode strings issue on SQLServer and SQLAzure

### PR DESCRIPTION
**Pull request by sseshachala**
Fixes the Unicode strings issue as mentioned in
http://support.microsoft.com/kb/239530

### CODE REVIEW

**Comment by infograf768** [comment]
This fix has been comfirmed on http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=27785

File: libraries/joomla/database/database/sqlsrv.php [hunk 1]
```diff
@@ -671,6 +671,20 @@ public function query()
 		return $this->cursor;
 	}
 	/**
```
Comment (chdemko) [inline libraries/joomla/database/database/sqlsrv.php:673]: code style: add blank line

File: libraries/joomla/database/database/sqlsrv.php [hunk 2]
```diff
@@ -671,6 +671,20 @@ public function query()
...
+	 * Method to quote and optionally escape a string to database requirements for insertion into the database.
```
Comment (chdemko) [inline libraries/joomla/database/database/sqlsrv.php:674]: Code style: add space before each *

File: libraries/joomla/database/database/sqlsrv.php [hunk 3]
```diff
@@ -671,6 +671,20 @@ public function query()
...
+	 * @return  string  The quoted input string.
+	 *
+	 * @since   11.1
+	 */
+	public function quote($text, $escape = true)
+	{
+		return 'N' . '\'' . ($escape ? $this->escape($text) : $text) . '\'';
+	}
+	/**
 	 * This function replaces a string identifier <var>$prefix</var> with the string held is the
```
Comment (chdemko) [inline libraries/joomla/database/database/sqlsrv.php:688]: code style: add blank line

File: libraries/joomla/database/database/sqlsrv.php [hunk 4]
```diff
@@ -671,6 +671,20 @@ public function query()
...
+	 * @param   string   $text    The string to quote.
+	 * @param   boolean  $escape  True (default) to escape the string, false to leave it unchanged.
+	 *
+	 * @return  string  The quoted input string.
+	 *
+	 * @since   11.1
+	 */
+	public function quote($text, $escape = true)
+	{
+		return 'N' . '\'' . ($escape ? $this->escape($text) : $text) . '\'';
```
Comment (chdemko) [inline libraries/joomla/database/database/sqlsrv.php]: You could use 'N\'' . ($escape ? $this->escape($text) : $text) . '\'';

File: libraries/joomla/database/database/sqlsrv.php [hunk 5]
```diff
@@ -671,6 +671,20 @@ public function query()
...
+	 * @param   boolean  $escape  True (default) to escape the string, false to leave it unchanged.
+	 *
+	 * @return  string  The quoted input string.
+	 *
+	 * @since   11.1
+	 */
+	public function quote($text, $escape = true)
+	{
+		return 'N\'' . ($escape ? $this->escape($text) : $text) . '\'';
+	}
```
Comment (elkuku) [inline libraries/joomla/database/database/sqlsrv.php:686]: Whitespace found at end of line

**Comment by chdemko** [comment]
@sseshachala The pull request cannot be merged.

**Comment by LouisLandry** [comment]
This has been sitting for over 2 months without update and is not currently mergeable.  I'm going to close it for now.  If you want to continue the discussion please rebase the branch against current master and re-open the pull request.  Cheers.

**Pull request closed without merging**